install_config/aggregate_logging.adoc

```diff
@@ -12,628 +12,482 @@ toc::[]
-in one user interface. There are two options for aggregating container logs,
-depending on user requirements:
+OpenShift cluster administrators can aggregate and view logs for a range of
+OpenShift services using the EFK stack, which is a modified version of the
+https://www.elastic.co/videos/introduction-to-the-elk-stack[ELK stack]. This
+topic shows you how to deploy the stack using a template.
 
-. link:#using-a-centralized-file-system[Using a centralized file system]
-. link:#using-elasticsearch[Using *Elasticsearch*]
+Three components make up the EFK logging stack:
 
-[IMPORTANT]
-====
-These solutions are a work in progress. As packaging improvements are made,
-these instructions will be simplified.
-====
-endif::[]
-ifdef::openshift-enterprise[]
-As an OpenShift administrator, you may want to view the logs from all containers
-in one user interface. The currently supported method for aggregating container
-logs in OpenShift Enterprise is link:#using-a-centralized-file-system[using a
-centralized file system]. Additional supported methods are planned for inclusion
-in future releases.
-
-[NOTE]
-====
-As packaging improvements are made, these instructions will be simplified.
-====
-endif::[]
+* Elasticsearch: an object store where all logs are fed.
+* Fluentd: gathers logs from nodes and feeds them to Elasticsearch.
+* Kibana: a web UI for Elasticsearch.
 
-[[using-a-centralized-file-system]]
+The EFK stack can be useful for tracking logs and analytics to help see how
+application users are using your application. For example, you can use
+https://www.elastic.co/products/elasticsearch[Elasticsearch] to store logs and
+transaction data, use http://www.fluentd.org/architecture[Fluentd] to collect
+the data, and use
+https://www.elastic.co/guide/en/kibana/current/introduction.html[Kibana] to
+filter and examine the data on the Kibana console.
 
-== Using a Centralized File System
+== Pre-deployment Configuration
 
-This method reads all container logs and forwards them to a central server for
-storage on the file system.
-ifdef::openshift-origin[]
-This solution requires less resources and requires less management than the
-link:#using-elasticsearch[*Elasticsearch* option], but the logs are not indexed
-and searchable.
-endif::[]
+Deploying the EFK stack requires:
 
-[[installing-fluentd-td-agent-on-nodes]]
-=== Installing fluentd (td-agent) on Nodes
+* A *logging-deployer* secret
+* A *logging-deployer* service account with the secret and create privileges
+* Sufficient volumes for Elasticsearch cluster storage
+* A router deployment for serving cluster-defined routes
 
-Perform the following steps on each node to install and configure *fluentd*
-(*td-agent*):
+The deployer generates the objects necessary for communication within the
+cluster and defines secrets and templates to implement aggregated logging.
 
-. Run the following commands:
+. Create a new project. Once implemented in a single project, the EFK stack will
```

Comment:
Since later steps say stuff like "for example, if your project was named `logging`", it would be nice if the example precedent was set at this project creation step.